SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,63 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
-
-        //public UserController() { }
-
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
-            var users = _context.Users.ToArray<User>();
-            return users;
+        public IEnumerable<User> Get()
+        {
+           return _context.Users.ToArray<User>();
         }
-
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
-            return user;
-        }
-
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
+        public async Task<User> Get(int id)
+        {
+            return await _context.Users.FindAsync(id);
         }
-
         // POST: api/User
-        [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        [HttpPost("{id}")]
+        public async Task<HttpResponseMessage> Post(int id,[FromBody] User user)
+        {
+            if (ModelState.IsValid)
+            {
+                //get user with the specified ID from database
+                User dbUser = await _context.Users.FindAsync(id);
+                //user not found
+                if(user == null)
```

Comment:
Should this not be dbUser == null?

By the looks of this post method. Its meant to create a user. If this is correct then your check currently (as in the code at the moment) for dbUser does nothing. The movelState.IsValid should already ensure that user is not null, so having an additional check for it should not be required, unless it was meant to ensure the user does not already exist

Reply:
oh thanks i missed dat

Reply:
fixing it now